evap/staff/forms.py

```diff
@@ -616,6 +626,19 @@ def show_delete_button(self):
 
         return False
 
+    def save(self, *args, **kwargs):
+        if self.instance.pk:
+            selected_questionnaires = self.cleaned_data.get("questionnaires")
+            removed_questionnaires = set(self.instance.questionnaires.all()) - set(selected_questionnaires)
+            TextAnswer.objects.filter(
+                contribution=self.instance, question__questionnaire__in=removed_questionnaires
+            ).delete()
+            RatingAnswerCounter.objects.filter(
+                contribution=self.instance, question__questionnaire__in=removed_questionnaires
+            ).delete()
```

Comment:
Should be maybe add some kind of database constraint to model the constraint that we violated here before?

> answers for (contribution, question) can only exist if question.questionnaire is included in contribution.questionnaires

I think it would be good to have it, but I don't know if it's feasable with the tools we have. There is quite some joining going on.

Reply:
seems quite complicated for a not really that important thing. but if you'd feel better with it, let's discuss that :)

Reply:
see #1728